napari/layers/base/base.py

```diff
@@ -318,57 +337,65 @@ def editable(self):
         return self._editable
 
     @editable.setter
-    def editable(self, editable):
-        if self._editable == editable:
-            return
-        self._editable = editable
-        self._set_editable(editable=editable)
-        self.events.editable()
+    def editable(self, value):
+        self.events.editable(value)
+
+    def _on_editable_change(self, value):
+        self._editable = value
+
+    def _update_editable(self):
+        self.editable = self._is_editable
```

Comment:
can you help me understand the flow here?  why do we need `editable`, `is_editable`,  and the indirection?

Reply:
yeah, this is a bit clunky, it really is because different layers can be editable in different configurations and so this allows layers to have some specific control over this. I don't love the pattern though ....

Reply:
This stuff also really isn't used by the image layer, but I had to change it a bit, as long as it's not broken (which I don't think it is) maybe we live with it for a little longer. Ultimately I don't think we want this stuff living on the layer model anyway (see #1353 etc for example)